### ISSUE TITLE
Role can fail on just one failed attempt to download repo, repo key, or specific Jenkins install

### ISSUE THREAD

**Issue by CloudSurgeon** [geerlingguy/ansible-role-jenkins#274]
Since all three of these mechanisms are pulling from external urls, a result->until-> retries pattern should be used. (the jenkins site was giving me fits today). 

`delphix-tcw-centos7-kitchen_sink: TASK [geerlingguy.jenkins : Ensure Jenkins repo is installed.] *****************
    delphix-tcw-centos7-kitchen_sink: changed: [default]
    delphix-tcw-centos7-kitchen_sink:
    delphix-tcw-centos7-kitchen_sink: TASK [geerlingguy.jenkins : Add Jenkins repo GPG key.] *************************
    delphix-tcw-centos7-kitchen_sink: fatal: [default]: FAILED! => {"changed": false, "msg": "failed to fetch key at https://pkg.jenkins.io/redhat/jenkins.io.key , error was: Request failed: <urlopen error timed out>"}
    delphix-tcw-centos7-kitchen_sink:
    delphix-tcw-centos7-kitchen_sink: PLAY RECAP *********************************************************************
    delphix-tcw-centos7-kitchen_sink: default       `

**Comment by geerlingguy** [comment]
Good idea; I have also run into this error plenty of times, as the Jenkins servers are notoriously flaky...

**Comment by CloudSurgeon** [comment]
Awaiting #279
